### PR TITLE
Fix Set MXID Dialog woes

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/4660
Signed-off-by: Michael Telatynski 7t3chguy@gmail.com

### CODE REVIEW

File: src/components/views/dialogs/SetMxIdDialog.js [hunk 1]
```diff
@@ -106,6 +106,15 @@ export default React.createClass({
     },
 
     _doUsernameCheck: function() {
+        // XXX: SPEC-1
+        // Check if username is valid
+        if (encodeURIComponent(this.state.username) !== this.state.username) {
+            this.setState({
+                usernameError: _t('User names may only contain letters, numbers, dots, hyphens and underscores.'),
```
Comment (t3chguy) [inline src/components/views/dialogs/SetMxIdDialog.js]: this string is copied from the registration page, so is already i18n'd

File: src/components/views/dialogs/SetMxIdDialog.js [hunk 2]
```diff
@@ -106,6 +106,15 @@ export default React.createClass({
     },
 
     _doUsernameCheck: function() {
+        // XXX: SPEC-1
+        // Check if username is valid
+        if (encodeURIComponent(this.state.username) !== this.state.username) {
```
Comment (lukebarnard1) [inline src/components/views/dialogs/SetMxIdDialog.js]: So is this needed because the username checking API sends the user name in the URI?

Comment (t3chguy) [inline src/components/views/dialogs/SetMxIdDialog.js]: It's used as a local validity check, synapse 500's if it sees utf8 on this endpoint and leaves the user confused, I'm just copying the logic from the registration page for the same endpoint

Comment (lukebarnard1) [inline src/components/views/dialogs/SetMxIdDialog.js]: Ah. Ok, probably worth a comment explaining that.

**Comment by lukebarnard1** [comment]
So it looks like with a comment on this, it should be good to go

**Comment by t3chguy** [comment]
@lukebarnard1 comment added

**Pull request merged**